src/org/zaproxy/zap/spider/parser/SpiderHtmlFormParser.java

```diff
@@ -242,6 +252,19 @@ private FormData prepareFormDataSet(FormFields form) {
 				currentList = submitFields;
 			}
 			for (String value : getValues(field)) {
```

Comment:
The submit fields should also be passed to the value generator, other than that (and the comment added previously) it looks good.